src/go/cmd/api_view.go

```diff
@@ -83,3 +100,20 @@ func createReview(pkgDir string) (PackageReview, error) {
 		PackageName: m.PackageName,
 	}, nil
 }
+
+func recursiveSortNavigation(n Navigation) {
+	for _, nn := range n.ChildItems {
+		recursiveSortNavigation(nn)
+	}
+	slices.SortFunc(n.ChildItems, func(a Navigation, b Navigation) int {
+		aa, err := json.Marshal(a)
```

Comment:
`Navigation` contains a map, should we take a similar approach as for `Diagnostic`?

Reply:
I don't think we need to. I checked the docs for the JSON marshaler and the maps are sorted by key. So the output for each `Navigation` should be deterministic. Unless I misunderstood the question?

Reply:
No, I was worried that the marshaler would make these comparisons unstable by iterating the maps. I didn't know it sorts them. I'm surprised to learn it does; naively, that seems like wasted effort.

Reply:
I was surprised too which is why I initially wrote the custom marshaler.

Here's the relevant docs taken from https://pkg.go.dev/encoding/json#Marshal

`
Map values encode as JSON objects. The map's key type must either be a string, an integer type, or implement [encoding.TextMarshaler](https://pkg.go.dev/encoding#TextMarshaler). The map keys are sorted and used as JSON object keys by applying the following rules, subject to the UTF-8 coercion described for string values above:
`

Reply:
Now I have to wonder whether json v2 behaves the same and what magnitude of perf improvement we could gain by not sorting maps 🤔